### DISCUSSION TITLE
Form.Control Input as textarea breaks onChange event

### DISCUSSION THREAD

**Discussion by jkanel** [rsuite/rsuite#2169] [Q&A]
**Does anyone know how to make an `Input ` "textarea" work correctly with the form onChange event in place?**

I'm experiencing an issue using the rsuite `<Form.Control accepter={Input} as="textarea" ...` inside a `Form`.

Specifically, the issue occurs when the form `onChange` event is triggered.   Each character typed into the `Input` control triggers the form `onChange` event and _appears to take focus out of the_ `Input`.  The result is that you can only type one character at a time and then have to reset focus in the `Input`.  **NOTE: This issue does not occur with regular `Input` (i.e., not "textarea").**  

Here's the sample code:
```
const Textarea = React.forwardRef((props, ref) => <Input {...props} as="textarea" ref={ref} />);

<Form
  formValue={formValue}
  checkTrigger="change"
  onChange={(value) => setFormValue(value)} 
  id={"form-".concat(primaryId)}
  fluid block
  readOnly={!userCanContribute} >

  <Form.Group controlId={"description-".concat(primaryId)}>
    <Form.Control accepter={Textarea} rows="5" block placeholder="Description" 
      defaultValue={rowData.description} name="description"  />
  </Form.Group>

</Form>
```

**Comment by simonguo** [comment]
hi @jkanel 
This situation is often caused by the form being re-rendered. You can reproduce your problem through [codesandbox](https://codesandbox.io/s/rsuite-template-5vq6zo2z5l) to see what the cause is.

**Reply by jkanel** [reply]
@simonguo - Here's the [codesandbox](https://codesandbox.io/s/clever-sun-5bro9?file=/src/index.js) link with a working example.  I look forward to hearing your ideas!  Thank you.

My thoughts...  
#1 - Standard Input is my preferred approach.  
#2 - looks fine, but has event issues  
#3 / #4 don't really suffice.

**Reply by simonguo** [reply]
hi @jkanel   
You have defined the `TextareaElement` component in the `App` component, which will cause the `TextareaElement` to be recreated every time the `App` renders. You should define `TextareaElement` independently. This is the principle of React.

```diff

+ const TextareaElement = React.forwardRef((props, ref) => (
+  <Input ref={ref} {...props} as="Textarea" />
+ ));

function App() {
  const [formValue, setFormValue] = React.useState({
    textAreaElement: "I lose focus with every key press",
    standard: "I work just fine.",
    titleCaseTextArea: "I lost my formatting",
    lowerCaseTextArea: "Uncomment my 'as' attribute and you'll see an error"
  });

-  const TextareaElement = React.forwardRef((props, ref) => (
-    <Input ref={ref} {...props} as="Textarea" />
-  ));

  return (
    ....
  )
}
```

**Comment by jkanel** [comment]
From @simonguo:

hi @jkanel
You have defined the TextareaElement component in the App component, which will cause the TextareaElement to be recreated every time the App renders. You should define TextareaElement independently. This is the principle of React.

+ const TextareaElement = React.forwardRef((props, ref) => (
+  <Input ref={ref} {...props} as="Textarea" />
+ ));

function App() {
  const [formValue, setFormValue] = React.useState({
    textAreaElement: "I lose focus with every key press",
    standard: "I work just fine.",
    titleCaseTextArea: "I lost my formatting",
    lowerCaseTextArea: "Uncomment my 'as' attribute and you'll see an error"
  });

-  const TextareaElement = React.forwardRef((props, ref) => (
-    <Input ref={ref} {...props} as="Textarea" />
-  ));

  return (
    ....
  )
}
